src/screens/Exchange/CoinifyWidget.js

```diff
@@ -206,35 +229,79 @@ export default function CoinifyWidget({
         scalesPageToFitmediaPlaybackRequiresUserAction
         automaticallyAdjustContentInsets={false}
         scrollEnabled={true}
-        style={{
-          flex: 0,
-          width: "100%",
-          height: "100%",
-        }}
+        style={styles.webview}
       />
-      {isWaitingDeviceJob ? (
-        <DeviceJob
-          deviceModelId="nanoX" // NB: EditDeviceName feature is only available on NanoX over BLE.
-          meta={meta}
-          onCancel={() => {
-            settleTrade("rejected");
-            setWaitingDeviceJob(false);
-          }}
-          onDone={() => {
-            settleTrade("accepted");
-            setWaitingDeviceJob(false);
-          }}
-          steps={[
-            connectingStep,
-            accountApp(mainAccount),
-            verifyAddressOnDeviceStep(mainAccount),
-          ]}
-        />
-      ) : null}
+      <BottomModal id="DeviceActionModal" isOpened={isOpen}>
+        <View style={styles.modalContainer}>
+          {isWaitingDeviceJob ? (
+            <DeviceAction
+              action={action}
+              device={device}
+              request={{ account: mainAccount, tokenCurrency }}
+              onResult={onResult}
+            />
+          ) : (
+            <VerifyAddress
+              account={mainAccount}
+              device={device}
+              onResult={onVerify}
+            />
+          )}
+        </View>
+      </BottomModal>
     </View>
   );
 }
 
+function VerifyAddress({
+  account,
+  device,
+  onResult,
+}: {
+  account: Account,
+  device: Device,
+  onResult: (confirmed: boolean, error?: Error) => void,
+}) {
+  const { t } = useTranslation();
+
+  const onConfirmAddress = useCallback(async () => {
+    try {
+      const { address } = await withDevice(device.deviceId)(transport =>
+        from(
+          getAddress(transport, {
+            derivationMode: account.derivationMode,
+            currency: account.currency,
+            path: account.freshAddressPath,
+            verify: true,
```

Comment:
we should now use the bridge to do a verify.

```
getAccountBridge(account)
      .receive(account, {
        deviceId: meta.deviceId,
        verify: true,
      })
```

this is critical for Bitcoin Cash which have different address format than default ones. If you look at master of LLM we should have migrated all getAddress to this.